### PR TITLE
Add function test names to email and url

### PR DESCRIPTION
**Pull request by aytee17**
I noticed that `string.email()` and `string.url()` don't have associated test function names and appear as `undefined` in the `tests` array of `SchemaDescription`. 

This just makes sure that a proper name is sent through the `matches` function.

### CODE REVIEW

**Comment by jquense** [comment]
why close this?

**Comment by aytee17** [comment]
Honestly, I thought I did something wrong (this is my first PR changing code) so I closed it out of embarrassment.

**Comment by dontub** [comment]
`matches` must have a name itself if none is provided, probably just "matches". Without name at least the async validation is (correctly) rejected, but without errors. Until this is fixed I have to overwrite the `matches` method.

File: src/string.js [hunk 1]
```diff
@@ -76,14 +76,20 @@ inherits(StringSchema, MixedSchema, {
...
+        options.message ||
+        options.hasOwnProperty('excludeEmptyString') ||
+        options.name
+      ) {
+        ({ excludeEmptyString, message, name } = options);
       } else message = options;
     }
 
     return this.test({
+      name,
```
Comment (jquense) [inline src/string.js]:
```suggestion
      name: name || 'matches'
```

**Comment by juice49** [comment]
@jquense do you have a schedule for publishing a new version of yup to npm?

I'm relying on this patch in one of my projects, so I've included the yup master branch for now. It would be great to be able to use yup from npm.

Happy to help if there is anything I can do 🙂.

Thank you for this very useful lib!

**Comment by sawyerh** [comment]
Thanks for tackling this @aytee17, I just noticed this too and was about to open a PR. Glad to see this got merged, and eagerly awaiting its release. Our system logs the `name` to help track what type of errors users are running into.

**Pull request merged**